src/cdk/overlay/position/flexible-connected-position-strategy.spec.ts

```diff
@@ -1119,6 +1119,102 @@ describe('FlexibleConnectedPositionStrategy', () => {
       expect(Math.floor(overlayRect.height)).toBe(OVERLAY_HEIGHT);
     });
 
+    it('should set the proper styles when the `bottom` value is exactly zero', () => {
+      originElement.style.position = 'fixed';
+      originElement.style.bottom = '0';
+      originElement.style.left = '200px';
+
+      positionStrategy
+        .withFlexibleWidth()
+        .withFlexibleHeight()
+        .withPositions([{
+          overlayY: 'bottom',
+          overlayX: 'start',
+          originY: 'bottom',
+          originX: 'start'
+        }]);
+
+      attachOverlay({positionStrategy});
+
+      const boundingBox = overlayContainer
+        .getContainerElement()
+        .querySelector('.cdk-overlay-connected-position-bounding-box') as HTMLElement;
+
+      expect(boundingBox.style.bottom).toBe('0px');
```

Comment:
Add comment for these tests like "Ensure that `0px` is explicitly set rather than the property being left blank due to zero being falsy."